### PR TITLE
Experiment: Apply ```#[deprecated_safe]``` to ```env::set_var```/```env::remove_var```

### PR DESCRIPTION
**Pull request by ghost**
cc #94978
cc #95025
cc #94619

### CODE REVIEW

**Comment by skippy10110** [comment]
@lcnr, i didn't think a draft PR would pick a reviewer 😅 this is just an experiment (so far) to see what applying ```#[deprecated_safe]``` within libstd might look like, the PR this is based on (#95025) hasn't landed yet

**Comment by lcnr** [comment]
will be away next week, so if it gets ready for review by that time, feel free to reassign this, will needs `libs-api` signoff anyways

File: library/test/src/term/terminfo/searcher/tests.rs [hunk 1]
```diff
@@ -13,7 +13,11 @@ fn test_get_dbpath_for_term() {
     }
     assert!(x("screen") == "/usr/share/terminfo/s/screen");
     assert!(get_dbpath_for_term("") == None);
+    // FIXME(skippy) there's no fix for deprecated_safe until tests can be run single threaded
+    #[cfg_attr(not(bootstrap), allow(deprecated_safe))]
```
Comment (riking) [inline library/test/src/term/terminfo/searcher/tests.rs:17]: This style of `cfg_attr`s is suspicious - aren't the bootstrap cfgs going to be deleted at the next upgrade?

Instead, use `// SAFETY: FIXME(skippy) This code is incorrect.`

**Comment by CAD97** [comment]
Aside: [`before_exec`](https://doc.rust-lang.org/nightly/std/os/unix/process/trait.CommandExt.html#method.before_exec) should probably also be marked `#[deprecated_safe]`.

**Pull request closed without merging**